app/Spago.hs

```diff
@@ -146,10 +146,11 @@ getAllDependencies Config { dependencies = deps, packages = pkgs } =
               Map.insert dep x newAcc
 
 -- | Fetch all dependencies into `.spago/`
-install :: Maybe Int -> IO ()
-install maybeLimit = do
+install :: Maybe Int -> [PackageName] -> IO ()
+install maybeLimit packages = do
   -- Make sure .spago exists
   T.mktree $ T.fromText spagoDir
+  addDependencies packages
```

Comment:
1. Agreed. My maybe implementation was checking that, but I let the simplest types in when I switched to a list :)
2. Will do, for some reason I was thinking I shouldn't do text effects in the addDependencies function, but the rest of the file does, so easy to add.

Though should we throw if the packages aren't in the package set?, I think we would be wanting to end the install quickly if something that the user is doing went wrong?

Reply:
The rule of thumb is that you should throw (i.e. interrupt the flow of the program) only when you're in the "most outer layer", and you're reasonably sure that your interruption will not cause inconsistent state in the outside world.

In this case we know that we're being called by other places, and we don't have enough context to decide that we can interrupt the flow.
E.g. this is being called by the code that ports your psc-package config, and that code might do other things after calling this, thus it should not be interrupted (if we want to signal this we should use the `MonadThrow` constraint, so that the caller knows that exceptions should be handled).

But I'd lean for having this just print a warning (since if we don't add more dependencies it's not a problem at all, as long as the user is warned about it), or if we want to signal to the caller an error condition (so that they can eventually throw if they want) we should return something like `Either SpagoError (IO ())` (returning the IO action in case of success, so that the caller can run it if everything is fine)

Reply:
Added in the check for packages list being empty.

Implemented your suggestion with the warning :)